source/server-discovery-and-monitoring/server-discovery-and-monitoring-tests.rst

```diff
@@ -143,3 +143,76 @@ Command response on an arbiter, recovering member, ghost, or secondary
 when slaveOk is false:
 
     {ok: 0, errmsg: "not master"}
+
+Network error on monitoring connection
+--------------------------------------
+
+Non-timeout network error
+'''''''''''''''''''''''''
+
+**Scenario:**
+
+Subscribe to `TopologyDescriptionChanged SDAM event`_ on the MongoClient.
+
+Subsribe to `PoolCleared CMAP event`_ on the MongoClient.
+
+Set a `failCommand fail point`_ on ``isMaster`` command
+with ``closeConnection: true`` parameter. The following pseudocode illustrates
+setting the fail point:
+
+.. code:: ruby
+
+    admin_client.command(
+      configureFailPoint: 'failCommand',
+      mode: {times: 2},
+      data: {
+        failCommands: %w(isMaster),
```

Comment:
Can we stick to JS and/or a shell example here? I'm not familiar with `%w` (literal array syntax?) but the more generic we can keep this the better.